src/engine/Server.cpp

```diff
@@ -131,8 +131,9 @@ void Server::run(const string& indexBaseName, bool useText, bool usePatterns,
 
   // First set up the HTTP server, so that it binds to the socket, and
   // the "socket already in use" error appears quickly.
-  auto httpServer = HttpServer{port_, "0.0.0.0", static_cast<int>(numThreads_),
```

Comment:
The HttpServer + Server architecture desperately needs an overhaul, and it shows here.
Both classes are doing similar things and therefore they share a couple of objects, but it's not ideal we need to pass them around like this